Doc/library/tempfile.rst

```diff
@@ -75,20 +75,50 @@ The module defines the following user-callable items:
-
-   This function operates exactly as :func:`TemporaryFile` does, except that
-   the file is guaranteed to have a visible name in the file system (on
-   Unix, the directory entry is not unlinked).  That name can be retrieved
-   from the :attr:`name` attribute of the returned
-   file-like object.  Whether the name can be
-   used to open the file a second time, while the named temporary file is
-   still open, varies across platforms (it can be so used on Unix; it cannot
-   on Windows).  If *delete* is true (the default), the file is
-   deleted as soon as it is closed.
+.. function:: NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, delete=True, *, errors=None, delete_on_close=True)
+
+   This function operates exactly as :func:`TemporaryFile` does, except the
+   following differences:
+
+   * The file is guaranteed to have a visible name in the file system (on Unix,
+     the directory entry is not unlinked).
+
+   * There is more granularity in the deletion behaviour of the file (see
+     ``delete_on_close`` below)
+
    The returned object is always a file-like object whose :attr:`!file`
-   attribute is the underlying true file object. This file-like object can
-   be used in a :keyword:`with` statement, just like a normal file.
+   attribute is the underlying true file object. This file-like object can be
+   used in a :keyword:`with` statement, just like a normal file.The name of the
+   temporary file can be retrieved from the :attr:`name` attribute of the
+   returned file-like object.
+
+   If *delete* is true (the default) and *delete_on_close* is true (the
+   default), the file is deleted as soon as it is closed. If *delete* is true
+   and *delete_on_close* is false, the file is deleted on context manager exit
+   only. If *delete* is false, the value of *delete_on_close* is ignored.
+
+   While the named temporary file is open, the file can always be opened again
+   on POSIX. On Windows, it can be opened again if *delete* is false, or if
+   *delete_on_close* is false, or if the additional open shares delete access
+   (e.g. by calling :func:`os.open` with the flag ``O_TEMPORARY``).  On
+   Windows, if *delete* is true and *delete_on_close* is false, additional
+   opens that do not share delete access (e.g. via builtin :func:`open`) must
+   be closed before exiting the context manager, else the :func:`os.unlink`
+   call on context manager exit will fail with a :exc:`PermissionError`.
+
+   To use the name of the temporary file to open the closed file second time,
+   either make sure not to delete the file upon closure (set the *delete*
+   parameter to be false) or, in case the temporary file is created in a
+   :keyword:`with` statement, set the *delete_on_close* parameter to be false.
+   The latter approach is recommended as it provides assistance in automatic
+   cleaning of the temporary file upon the context manager exit.
```

Comment:
The previous paragraph explain the basic operation of the `delete` and delete_on_close` parameters and that the file can be opened again if `delete` or `delete_on_close` is false. I was trying to avoid restating this because it's getting kind of long-winded.

Reply:
I think one can work out this information from the previous pragagraphs, but from the not so advanced user  point of view (like myself) I would want to have this paragraph as well, as it describes arguably the most frequent use of the named temporary file: write some data, close it and then use its name to open it further.